ui/src/app/store/machine/selectors.ts

```diff
@@ -273,6 +277,27 @@ const getByStatusCode = createSelector(
     machines.filter(({ status_code }) => status_code === statusCode)
 );
 
+/**
+ * Get the deployed machines with the provided tag.
+ * @param state - The redux state.
+ * @param tagId - The tag id.
+ * @returns The deployed machines with the tag.
+ */
+const getDeployedWithTag = createSelector(
+  [
+    defaultSelectors.all,
+    (_state: RootState, tagId: Tag[TagMeta.PK] | null) => tagId,
+  ],
+  (machines, tagId) => {
+    if (!isId(tagId)) {
+      return [];
+    }
+    return machines
+      .filter(({ status }) => status === NodeStatus.DEPLOYED)
+      .filter(({ tags }) => tags.includes(tagId));
```

Comment:
Could this be handled in a single `filter`?

Reply:
Hah, yes totally, thanks!